Classes/Emogrifier.php

```diff
@@ -1289,41 +1292,71 @@ function (array $matches) {
         $trimmedLowercaseSelector = trim($lowercasePaddedSelector);
         $xPathKey = md5($trimmedLowercaseSelector);
         if (!isset($this->caches[self::CACHE_KEY_XPATH][$xPathKey])) {
-            $roughXpath = '//' . preg_replace(
-                array_keys($this->xPathRules),
-                $this->xPathRules,
-                $trimmedLowercaseSelector
-            );
-            $xPathWithIdAttributeMatchers = preg_replace_callback(
-                self::ID_ATTRIBUTE_MATCHER,
-                [$this, 'matchIdAttributes'],
-                $roughXpath
-            );
-            $xPathWithIdAttributeAndClassMatchers = preg_replace_callback(
-                self::CLASS_ATTRIBUTE_MATCHER,
-                [$this, 'matchClassAttributes'],
-                $xPathWithIdAttributeMatchers
-            );
+            $notSelector = null;
+            if (strstr($trimmedLowercaseSelector, ':not')) {
```

Comment:
According to http://php.net/strstr , using strpos !== false is faster for finding out whether a string has a certain substring.